File: python3/Math and Geometry/reverse_32_bit_integer.py
import math


def reverse_32_bit_integer(n: int) -> int:
    INT_MAX = 2**31 - 1
    INT_MIN = -2**31
    reversed_n = 0
    # Keep looping until we've added all digits of 'n' to 'reversed_n' 
    # in reverse order.
    while n != 0:
        # digit = n % 10
        digit = int(math.fmod(n, 10))
        # n = n // 10
        n = int(n / 10)
        # Check for integer overflow or underflow.
        if (reversed_n > int(INT_MAX / 10) 
            or reversed_n < int(INT_MIN / 10)):
            return 0
        # Add the current digit to 'reversed_n'.
        reversed_n = reversed_n * 10 + digit
    return reversed_n

File: cpp/Math and Geometry/reverse_32_bit_integer.cpp
#include <climits> 

int reverse32BitInteger(int n) {
    int reversedN = 0;
    // Keep looping until we've added all digits of 'n' to 'reversedN' 
    // in reverse order.
    while (n != 0) {
        // Extract the last digit of 'n'.
        int digit = n % 10;
        // Remove the last digit from 'n'.
        n /= 10;
        // Check for integer overflow or underflow.
        if (reversedN > INT_MAX / 10 || reversedN < INT_MIN / 10) {
            return 0;
        }
        // Add the current digit to 'reversedN'.
        reversedN = reversedN * 10 + digit;
    }
    return reversedN;
}